lib/project_types/extension/models/specification_handlers/checkout_ui_extension.rb

```diff
@@ -4,12 +4,23 @@ module Extension
   module Models
     module SpecificationHandlers
       class CheckoutUiExtension < Default
+        L10N_FILE_SIZE_LIMIT = 16 * 1024 # 16kb
+        L10N_BUNDLE_SIZE_LIMIT = 256 * 1024 # 256kb
+        LOCALE_CODE_FORMAT = %r{
+          \A
+          (?<language>[a-z]{2,3}) # Language tag
+          (?:
+           -
+           (?<region>[a-zA-Z0-9]+) # Optional region subtag
```

Comment:
Do we want to enforce uppercase two-letter [region codes](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2)?

Reply:
Or were you thinking of doing it in core?

Reply:
I suggest we strictly enforce our expectation of the region code until we find out that's a problem (I can't think of why it would be). 

Your suggestion of enforcing that on the backend seems fine. If that's the case, could you add a comment here, something like "a more strict regex will be applied on the server" ?

Reply:
Adjusted the locale regex after our discussions to follow the spec of:
_Each filename consists of a [BCP 47 language tag](https://en.wikipedia.org/wiki/IETF_language_tag) with an optional [ISO 3166-1 region code](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2)_

We're only doing a regex in the CLI code and not trying to interpret whether the letters represent a known language or region.
Also updated it to basically ignore lowercase/uppercase checks. We will adjust the formatting to what we desire it to be (`fr-CA` instead of `fr-ca` or `FR-CA`) in the backend.